cloud/auth/auth.go

```diff
@@ -139,17 +140,16 @@ func requestToken(authConfig astro.AuthConfig, verifier, code string) (Result, e
 
 func authorizeCallbackHandler() (string, error) {
 	m := http.NewServeMux()
-	s := http.Server{Addr: "localhost:12345", Handler: m, ReadHeaderTimeout: 0}
+	s := http.Server{Addr: callbackServer, Handler: m, ReadHeaderTimeout: 0}
 	m.HandleFunc("/callback", func(w http.ResponseWriter, req *http.Request) {
 		defer req.Body.Close()
 		if errorCode, ok := req.URL.Query()["error"]; ok {
-			log.Fatalf(
-				"Could not authorize your device. %s: %s",
-				errorCode, req.URL.Query()["error_description"],
-			)
+			callbackChannel <- CallbackMessage{errorMessage: fmt.Sprintf("Could not authorize your device. %s: %s",
+				errorCode, req.URL.Query()["error_description"])}
+			resp := &http.Request{}
+			http.Redirect(w, resp, "https://auth.astronomer.io/device/denied", http.StatusFound)
```

Comment:
nit: might want to make `https://auth.astronomer.io/device/denied` & `https://auth.astronomer.io/device/success` a constant